src/openrct2-ui/windows/Themes.cpp

```diff
@@ -691,18 +703,27 @@ static WindowClass window_themes_tab_7_classes[] = {
 
         void OnScrollMouseDown(int32_t scrollIndex, const ScreenCoordsXY& screenCoords) override
         {
-            if (screenCoords.y / _row_height < GetColourSchemeTabCount())
+            if (screenCoords.y / _max_row_height < GetColourSchemeTabCount())
             {
-                int32_t y2 = screenCoords.y % _row_height;
-                _colour_index_1 = screenCoords.y / _row_height;
-                _colour_index_2 = ((screenCoords.x - _button_offset_x) / 12);
+                _colour_index_1 = GetColourIndex(screenCoords.y);
 
                 WindowClass wc = GetWindowClassTabIndex(_colour_index_1);
                 int32_t numColours = ThemeDescGetNumColours(wc);
+
+                // add colour to account for text wrapping
```

Comment:
```suggestion
                // If the window description wraps, make sure the row is high enough.
```

Reply:
Also, I see this pattern multiple times in you changes. To me, it seems you could extract that out into a (static) function.